micro_dl/cli/image_inference.py

```diff
@@ -229,33 +261,64 @@ def run_prediction(args, gpu_ids, gpu_mem_frac):
                     np.save(file_name, im_pred, allow_pickle=True)
                 else:
                     raise ValueError('Unsupported file extension')
+
+                # assuming target and predicted images are always 2D for now
+                # Load target
+                meta_idx = aux_utils.get_meta_idx(
+                    frames_meta,
+                    time_idx,
+                    target_channel,
+                    slice_idx,
+                    pos_idx,
+                )
+                file_path = os.path.join(
+                    args.image_dir,
+                    frames_meta.loc[meta_idx, "file_name"],
+                )
+                # get a single row of frame meta data
+                test_frames_meta_row = frames_meta.loc[[meta_idx]]
+                im_target = preprocess_imstack(
+                    frames_metadata=frames_meta,
+                    input_dir=args.image_dir,
+                    depth=1,
+                    time_idx=time_idx,
+                    channel_idx=target_channel,
+                    slice_idx=slice_idx,
+                    pos_idx=pos_idx,
+                )
+                im_target = image_utils.crop2base(im_target)
+                if data_format == 'channels_first':
```

Comment:
preprocess_imstack adds new axis at axis=2, by default assembles in xyz format, the check here should be on image_format I believe?

Reply:
This is my bad, from when I used data and image format interchangeably. Now it should be if image_format == 'zyx'.

Reply:
I saw that the preprocessing config switched to 'image_format', 'data_format' is still used in the network config. Here the input image format has to match the the network input format. Should we use 'image_format' for both preprocessing and network configs?

Reply:
They are not interchangeable, data_format refers to channel location and image_format refers do the order of the 3 dimensions (default zyx).
The training currently gets the image_format from the preprocessing config, so maybe we should add it to the network config too since that's the only one being read here...
https://github.com/czbiohub/microDL/blob/master/micro_dl/cli/train_script.py#L199
For now it can default to assuming image_format = zyx.